config/initializers/comfy_patching.rb

```diff
@@ -29,7 +29,19 @@ def assign_layout_categories
         end
       end
 
-      delete_orphan_categories && return if _categories.blank?
+      # Two situations for deleting orphan categories, either _categories has no or a restricted set of the currently existing categories
+      # Or, _categories consists of tags with entirely new labels, and hence new layouts would have to be created
+      # Not handled at present - adding new categories which equates to adding new layout categories which means _layout_category is empty
+      # _categories = [{ tag_params: 'topics', tag_class: 'categories'}]
+      # layouts_categories = [{label: 'topics'...}, {label: 'types'...}]
+      # orphan_categories = [{label: 'types'}]
+      layouts_categories = Comfy::Cms::LayoutsCategory.where(layout_id: self.id)
+      .map(&:layout_category)
+      _categories.each do |_cat| 
+        layouts_categories.filter! { |category| _cat[:tag_params] != category.label }
```

Comment:
Care this is not actually deleting layouts categories. This means that even if you end up with a `layouts_categories` array of one element only, the `delete_orphan_categories` is still fetching all `layouts_categories` of the given layout regardless.

Example. If you have 2 layouts categories and remove one, `delete_orphan_categories` will remove both anyway

Reply:
It is deleting the filtered `layouts_categories`, not the original `layouts_categories` that is defined in line 38.

Reply:
Line 67 in the `delete_orphan_categories` is fetching all layouts_categories of the layout, not the local variable defined in this method, exactly because `layouts_categories` in line 38 is a local variable only visible within `assign_layout_categories`.

So line 67 is basically doing `self.layouts_categories.map(&:destroy)` , which can also be seen as `Comfy::Cms::LayoutsCategory.where(layout_id: self.id).map(&:destroy)`

Perhaps it would also be good to rename the local variable on line 38 to avoid confusion?

Please give this a test again on your local environment yourself to ensure this is actually how it's behaving.
Given 2 categories and deleting 1, it results in the `Comfy::Cms::LayoutsCategory.where(layout_id: the_layout_id)` to be empty for me.

Reply:
Sorry, you're right, the way I evaluated this was a simple eye test to see whether the Types filters were removed from the News and Stories page if I deleted the types tag, and they were. I now realise that there's more to it especially if we were to export and import the updated News and Stories layout

Reply:
OK, I believe it was because it was not creating the new LayoutsCategory instances after it deleted the categories as well - it was just immediately returning and not proceeding to the `find_or_create_by` loop